DuckDuckGo/OnboardingExperiment/Manager/OnboardingManager.swift

```diff
@@ -75,27 +92,33 @@ extension OnboardingManager: OnboardingHighlightsManaging, OnboardingHighlightsD
 // MARK: - Add to Dock
 
 protocol OnboardingAddToDockManaging: AnyObject {
-    var isAddToDockEnabled: Bool { get }
+    var addToDockEnabledState: OnboardingAddToDockState { get }
 }
 
 protocol OnboardingAddToDockDebugging: AnyObject {
-    var isAddToDockLocalFlagEnabled: Bool { get set }
+    var addToDockLocalFlagState: OnboardingAddToDockState { get set }
     var isAddToDockFeatureFlagEnabled: Bool { get }
 }
 
 extension OnboardingManager: OnboardingAddToDockManaging, OnboardingAddToDockDebugging {
 
-    var isAddToDockEnabled: Bool {
-        // TODO: Add variant condition once the experiment is setup
-        isIphone && isAddToDockLocalFlagEnabled && isAddToDockFeatureFlagEnabled
+    var addToDockEnabledState: OnboardingAddToDockState {
+        // TODO: Add variant condition OR local conditions
+        if isAddToDockFeatureFlagEnabled && isIphone && addToDockLocalFlagState == .contextual {
+            return .contextual
+        } else if isAddToDockFeatureFlagEnabled && isIphone && addToDockLocalFlagState == .intro {
+            return .intro
+        } else {
+            return .disabled
+        }
```

Comment:
maybe could have a guard statement
guard isAddToDockFeatureFlagEnabled && isIphone else { return .disable }
and then check based on the addToDockLocalFlagState